File: jass-to-ts/src/main/java/org/vsx/jass/Token.java
package org.vsx.jass;

/** Токен */
public class Token {
    public String getType() {
        return TokenKind.GetType(Kind);
    }
    public String Kind = "";
    public int Line = 0;
    public int Col = 0;
    public int Pos = 0;
    public String Text = "";

    public Token() {}
    public Token(int col, int line, int pos, String text, String kind) {
        Col = col;
        Line = line;
        Pos = pos;
        Text = text;
        Kind = kind;
    } 
    
    @Override
    public String toString() {
        return String.format("%s,%s [%s|%s]: %s", Line, Col, getType(), Kind, Text);
    }

    public Token Clone() {
        return new Token(Col, Line, Pos, Text, Kind);
    }
}

File: JassToTsRust/src/Token.rs
pub struct Token {
    //Type => TokenKind.GetType(Kind),
    Kind: String,
    Line: i8,
    Col: i8,
    Pos: i8,
    Text: String,
}

impl Token {
    pub fn new(kind: String, line: i8, col: i8, pos: i8, text: String) -> Token {
        Token {
            Kind: kind,
            Line: line,
            Col: col,
            Pos: pos,
            Text: text,
        }
    }
    //public override string ToString() => $"{Line},{Col} [{Type}|{Kind}]: {Text}";
    //public Token Clone() => new Token { Kind = Kind, Line = Line, Col = Col, Pos = Pos, Text = Text };
}